src/main/java/org/fenixedu/academic/domain/DegreeCurricularPlan.java

```diff
@@ -1864,6 +1865,10 @@ public boolean canSubmitImprovementMarkSheets(final ExecutionYear executionYear)
         return sortedExecutionDegrees.last().getExecutionYear().equals(executionYear.getPreviousExecutionYear());
     }
 
+    public ExecutionInterval getBegin() {
+        return Collections.min(getBeginContextExecutionYears(), ExecutionYear.COMPARATOR_BY_YEAR);
```

Comment:
Beware that if the DCP has no associated execution year, this will throw a `NoSuchElementException`, instead of returning null.